java/arcs/core/host/Allocator.kt

```diff
@@ -121,20 +122,20 @@ class Allocator(val hostRegistry: HostRegistry) {
                     }
                 PlanPartition(
                     arcId.toString(),
-                    it.key::class.java.canonicalName!! /* ArcHost */,
+                    it.key.hostId(),
                     handleConnectionSpecs
                 )
             }
 
     /**
-     * Find [ArcHosts] by looking up known registered particles in every [ArcHost],
-     * mapping them to fully qualified Java classnames, and comparing them with the
-     * [ParticleSpec.location].
+     * Find [ArcHosts] by looking up known registered particles in every [ArcHost] by their
+     * [ParticleIdentifier] derived from [ParticleSpec.location].
      */
     private suspend fun findArcHostBySpec(spec: ParticleSpec): ArcHost =
-        hostRegistry.availableArcHosts()
-            .filter { host ->
-                host.registeredParticles().map { clazz -> clazz.java.getCanonicalName() }
-                    .contains(spec.location)
-            }.firstOrNull() ?: throw ParticleNotFoundException(spec)
+        hostRegistry.availableArcHosts().firstOrNull { host ->
+            runBlocking {
```

Comment:
Does this need to `runBlocking`? the functionn `findArcByHostId` is `suspend`

Reply:
removed